---
title: "How to talk to your bioinformatician?"
author: 
  - name: "January Weiner"
    orcid: 0000-0003-1438-7819
    affiliations:
     - ref: cubi
affiliations:
  - id: cubi
    name: Core Unit for Bioinformatics, BIH@Charité
    address: Charitéplatz 1, 
    postal-code: 10117 
    city: Berlin 
    country: Germany
title-slide-attributes:
  data-background-image: images/schloss.jpg
  #data-background-image: files/bih_bg_logo.png
format:
  revealjs:
    footer: "Core Unit for Bioinformatics, BIH@Charite"
    theme: cubi.scss
    logo: files/bih_logo_small.png
    transition: fade
    slide-number: "c/t"
    smaller: true
    navigation-mode: linear
    self-contained: true
comments:
  hypothesis: true
knitr:
  opts_chunk:
    dev: "svg"
---


## Then and now...

:::: {.columns}
::: {.column width="45%"}

### How it started

![](images/kuba-rower-1.jpg){width="70%"}

:::
::: {.column width="10%"}
:::

::: {.column width="45%"}
### How it is going

![](images/P1030267.jpg){width="70%"}

:::
::::

## CUBI - Core Unit for Bioinformatics

:::: {.collumns}
::: {.column width="45%"}

### Berlin Institute of Health:

 * Institute for translational and precision medicine
 * Founded in 2013 by MDC and Charité 

### Core Unit for Bioinformatics (CUBI):

 * Bioinformatics services and training to
   researchers at BIH, MDC and Charité
 * Focus on high throughput data analysis, user empowerment and
   reproducible research
 * High performance computing infrastructure, long term large data storage 

:::
::: {.column width="10%"}
:::
::: {.column width="45%"}

![](images/charite.jpg)


:::
::::

 

## What is this thing we call "bioinformatics"?

```{scss}
$callout-color-tip: #70ADC1;
```

```{css, echo=FALSE}
.large {
  font-size: 1.5em !important;
  background-color: "#481567" !important;
  color: white !important;
}

.theme-light .mermaid {
    --color-red: #ff7777;
    --color-orange: #ffbb99;
    --color-yellow: #ffeeaa;
    --color-green: #bbffbb;
    --color-cyan: #bbffee;
    --color-blue: #aaccff;
    --color-purple: #ddbbff;
    --color-pink: #ffcccc;
}

.viri1 {
  background-color: "#481567";
  color: white;
}

.whosaid {
  text-align: right !important;
}

.aside {
  font-size: 0.65em !important;
}

.callout {
  font-size: 1.1em !important;
}
```


```{r echo=FALSE,warning=FALSE,message=FALSE}
#| label: setup
library(tidyverse)
library(ggplot2)

theme_set(theme_minimal())
```




```{dot}
#| fig-width: 12
graph G {
  layout = "neato"
  fontname = "Helvetica,Arial,sans-serif"
  scale = 1.2

  node [
		style = filled
		color = "#000000"
		fontname = "Helvetica,Arial,sans-serif"
    fontsize = 10
    shape = "box"
	]

  statistics [ fillcolor="#F8B229", style=filled, label="Statistics", fontsize=18 ]
  bioinformatics [ fontcolor="white", fillcolor="#481567", style=filled, label="Bioinformatics", fontsize=18 ]
  cbio [ fontcolor="white", fillcolor="#33638d", style=filled, label="Computational\nbiology", fontsize=18 ]

  "Computers and science" [ fontcolor="white", fontsize=20, label = "Computers\nand\nScience" ]

  "Computers and science" -- bioinformatics
  "Computers and science" -- cbio
  "Computers and science" -- statistics


  subgraph Bioinformatics {
    node [ style=filled, fillcolor="#481567cc", fontcolor="white", fontsize=14 ]
    bioinformatics -- algorithms -- implementation
    bioinformatics -- formalizations
    bioinformatics -- standards
    bioinformatics -- databases -- maintenance
  }

  subgraph cbio {
      node [ fontcolor="white", fillcolor="#33638dcc", style=filled, fontsize=14 ]
      cbio -- "data\nscience" -- "data\nstewardship"
      cbio -- "functional\ngenomics"
      "functional\ngenomics" -- "gene set\nenrichment"
      cbio -- "using tools"
      cbio -- "sequence\nanalysis" -- "phylogenies"
  }

  subgraph Statistics {
      node [ fillcolor="#F8B22999cc", style=filled, fontsize=14 ]
      statistics -- "hypothesis\ntesting"
      "hypothesis\ntesting" -- "p-values"
      "hypothesis\ntesting" -- "effect sizes"
      statistics -- "power analysis"
      statistics -- "linear\nmodeling" -- "hierarchical\nmodeling"
      statistics -- "visualizations"
      statistics -- "Bayesian\nstatistics"
  }

}
```







::: {.notes}

I am not here to teach you immunology - or even bioinformatics. 
You will have to work with bioinformaticians, statisticians, computational biologists
or data scientists at some point in your career. I want to help you make
this collaboration as smooth as possible.

I also want to tell you a few things that bioinformaticians wish you knew.

 * What you call "bioinformatics" are different areas of science
 * We can't know all biology, so be patient with us
 * Bioinformatics is not a service
 * Bioinformatics is a scientific collaboration 

Be patient with us. We are not stupid, but we don't know your context, your
field, your abbreviations.

Also, we don't even know all of statistics or bioinformatics.

:::

## Statistics

![](images/statistics.png)


::: {.notes}

I am not her to to tell you that statistics is important. It is. But what I
really want to convey is that statistics is *a language*. You need to speak
this language if you want to communicate with other scientists.

The point of the next slide is not to show that you don't know statistics,
but that statistics is counter-intuitive.
:::

## What is a p-value?

$H_0$: The null hypothesis, no effect

$H_1$: The alternative hypothesis, there is an effect

We run a test, we get a p-value, say $0.03$. It is a probability.

Probability of *what*, exactly?


 1. Probability that $H_0$ is true (probability that there is no difference), given the data

 2. Probability that $H_1$ is true (probability that there is a difference), given the data

 3. Probability that the data is random

 4. Probability that the observations are due to random chance

 5. Probability of getting the same data by random chance

 . . .

 * **Probability of observing an effect at least as extreme given that $H_0$ is true**

::: {.notes}
None of that is true.

P-values are not what we intiutively think they are. 

They are in fact way less useful than we think they are.

The $H_0$ is true or false, there is no probability involved.
:::

## Our intuition is bayesian, not frequentist


| **Frequentist Statistics**    | **Bayesian Statistics**     |
|-------------------------------|-----------------------------|
| ![](images/classicalgiants.png)| ![](images/thomas-bayes.png)      |
| 1. Probability is defined as the long-run frequency of events | 1. Probability represents a degree of belief or certainty about an event |
| 2. Parameters (like the "true value") are fixed but unknown quantities. | 2. Parameters are treated as random variables with their own probability distributions. |
| 3. Asking about the probability of a hypothesis does not make sense | 3. Asking about the probability of a hypothesis is the main goal |

::: {.aside}

Ask me later to show you the coin trick! I can demonstrate to you that our
intuition is bayesian, not frequentist.

:::

## Why is that important?


:::: {.columns}

::: {.column width="55%"}

::: {.callout-tip}
## P-values are part of scientific language

 * *Always* use effect sizes 
 * Never rely on p-values alone

Know their limits: 

 * they control only type I errors (false positives)
 * they **do not** control type II errors (false negatives)

:::
:::

::: {.column width="5%"}
:::

::: {.column width="40%"}

![](images/dumbledore_p.jpg)

:::
::::

::: {.aside}
Wasserstein RL, Lazar NA. The ASA statement on p-values: context, process, and purpose. The American Statistician. 2016 Apr 2;70(2):129-33.
:::


## How Venn diagrams can fool scientists

COVID-19 study, both COVID-19 patients and non-COVID-19 patients are
compared in two groups of people, *G1* and *G2*. 

We wanted to know whether the influence of COVID-19 is different in these
two groups.

![](images/fig1-1.jpeg)


::: {.aside}
Venn diagrams may indicate erroneous statistical reasoning in
transcriptomics. Weiner, Obermayer and Beule, 2022, Frontiers Genetics
:::

## The results are artifacts!

Groups G1 and G2 were randomly drawn from the same population. They were
not different at all.

![](images/fig2-1.jpeg)


## What happens is, we are comparing significance with non-significance

> The Difference Between “Significant” and “Not Significant” is not
> Itself Statistically Significant

::: {.whosaid}
   (*Andrew Gelman and Howard Stern*)
:::

If a gene is significant in one comparison, and not significant in another,
that does not mean that there is a difference between the two groups.

It simply means that we *failed* to detect the difference in one of the
comparisons, but that is actually quite likely to happen!

::: {.callout-tip}
## Therefore:
Don't say "there is no difference". Say "we did not detect a difference".
:::

::: {.aside}
Gelman A, Stern H. The difference between “significant” and “not significant” is not itself statistically significant. The American Statistician. 2006 Nov 1;60(4):328-31.
:::


## Tale of two papers

![](images/pnas_1.png)


## Tale of two papers

![](images/pnas_res_1.png)


## Tale of two papers

![](images/pnas_1.png)

. . .

![](images/pnas_2.png)

## Tale of two papers

![](images/pnas_1.png)


![](images/pnas_2_red.png)

## Tale of two papers

![](images/pnas_res_2.png)

::: {.notes}
Key difference - second paper specifically focused on genes that were
regulated in either mouse model or human data, and asked whether they are
similar or not
:::

## Lessons learned

 * *A lot* depends on how you analyze your data

 . . .

 * This in turn depends on the questions you ask

 . . .

 * The average "Methods" section is not sufficient for reproducible
   science!

  . . .

Attempt to replicate 53 high-impact cancer biology papers:

> " Second, **none of the 193 experiments** were described in sufficient detail
> in the original paper to enable us to design protocols to repeat the
> experiments, so we had to seek clarifications from the original authors."
> (Errington et al., 2021)

::: {.aside}
Errington TM, Mathur M, Soderberg CK, Denis A, Perfito N, Iorns E, Nosek BA. Investigating the replicability of preclinical cancer biology. Elife. 2021 Dec 10;10:e71601.

Errington TM, Denis A, Perfito N, Iorns E, Nosek BA. Challenges for assessing replicability in preclinical cancer biology. Elife. 2021 Dec 7;10:e67995.
:::


## Excel and gene names

::: {.r-stack}
![](images/excel_paper_1.png){.fragment .absolute top=90 left=10 width=80%}

![](images/excel_hgnc.png){.fragment .absolute top=120 left=50 width=80%}

![](images/excel_paper_2.png){.fragment .absolute top=220 left=90 width=80%}
:::


::: {.aside style="font-size: 0.8em; color:red;"}
::: {style="font-size: 0.5em;"}
Ziemann, Mark, Yotam Eren, and Assam El-Osta. "Gene name errors are widespread in the scientific literature." Genome biology 17 (2016): 1-3; 
Abeysooriya M, Soria M, Kasu MS, Ziemann M. Gene name errors: Lessons not learned. PLoS Computational Biology. 2021 Jul 30;17(7):e1008984.
:::
:::

::: {.notes}
2016 paper: 20% of papers have Excel gene name errors
2021 paper: it is 30% now. Excel now learned mulitple languages
:::

## Bottom lines

:::: {.columns}
::: {.column width="40%"}

 * Communication is key
 * Reproducibility matters
 * Don't be like this ->
 * Learn how to program ;-)

::: 
::: {.column width="5%"}
:::
::: {.column width="55%"}

![](images/Last-line-of-defense---statistics.png){width="100%"}

:::
::::

## Thank you {.inverse background-color="#000000"}

:::: {.columns}

::: {.column width="40%"}
You can find a longer presentation along its source code 
at [https://github.com/bihealth/howtotalk](https://github.com/bihealth/howtotalk)

Parts of this have been expanded to a longer text which can be found at
[https://bihealth.github.io/howtotalk-book/](https://bihealth.github.io/howtotalk-book/)

A 5 day R crash course book is available at 
[https://bihealth.github.io/RCrashcourse-book/](https://bihealth.github.io/RCrashcourse-book/)

:::

::: {.column width="10%"}
:::

::: {.column width="50%"}

```{r}
#| label: qr
#| fig-width: 4.5
#| fig-height: 4.5
library(qrcode)
plot(qr_code("https://github.com/bihealth/howtotalk"))
```



:::
::::


## The statistical testing roulette

![](images/salmon_1.png)


## The statistical testing roulette

:::: {.columns}
::: {.column width="45%"}

**Subject.** One mature Atlantic Salmon (Salmo salar) participated in the fMRI study.
The salmon was approximately 18 inches long, weighed 3.8 lbs, and was not alive at
the time of scanning.

**Task.** The task administered to the salmon involved completing an open-ended
mentalizing task. The salmon was shown a series of photographs depicting human
individuals in social situations with a specified emotional valence. The salmon was
asked to determine what emotion the individual in the photo must have been
experiencing.

:::
::: {.column width="10%"}
:::
::: {.column width="45%"}

![](images/salmon_2.png)

:::
::::

::: {.aside}
Bennett CM, Miller MB, Wolford GL. Neural correlates of interspecies perspective taking in the post-mortem Atlantic Salmon: An argument for multiple comparisons correction. Neuroimage. 2009 Jul 1;47(Suppl 1):S125.
:::
















## The error is widespread

Nieuwenhuis et al. found that half of the scientists who could have
commited this error, did in fact commit this error.

![](images/erroneous_analyses.webp)

::: {.aside}
Nieuwenhuis, Sander, Birte U. Forstmann, and Eric-Jan
Wagenmakers. "Erroneous analyses of interactions in neuroscience: a problem
of significance." Nature neuroscience 14.9 (2011): 1105-1107.
:::


## Will "AI" change the field?

:::: {.columns}

::: {.column width="60%"}

![](images/statistics_ai_ml.jpeg)

:::
::: {.column width="5%"}
:::


::: {.column width="35%"}

 * New deep learning methods are useful, but hard to use 
 * Some of them are truly revolutionizing the field
 * There is still place for simpler ML algorithms
 * Ready to use LLMs (ChatGPT & Co.) have their use, but also limitations

:::
::::




